ml/git-log-to-jsonl

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import git
+import json
+import shutil
+from pathlib import Path
+
+
+def decode_bytes(data: bytes) -> str:
+    """
+    Attempts to decode bytes using a list of common encodings.
+    """
+    encodings_to_try = [
+        "utf-8",
+        "latin-1",
+        "iso-8859-1",
+        "cp1252",
+        "ascii",
+    ]
+
+    for encoding in encodings_to_try:
+        try:
+            return data.decode(encoding)
+        except UnicodeDecodeError:
+            continue
+
+    # If all else fails, use a replacement handler to avoid crashing
+    return data.decode("utf-8", errors="replace")
+
+
+def get_git_log_as_jsonl(repo_path: str):
+    """
+    Opens a local git repository, converts its log to a JSONL file.
+    """
+    repo_dir = Path(repo_path)
+
+    # Check if the path is a valid git repository
+    if not (repo_dir / ".git").exists():
+        print(f"Error: '{repo_path}' is not a valid git repository.")
+        return False
+
+    print(f"Opening repository at {repo_path}...")
+    try:
+        repo = git.Repo(repo_path)
+    except git.exc.InvalidGitRepositoryError:
+        print(f"Error: Path '{repo_path}' does not contain a valid git repository.")
+        return False
+
+    print("Generating JSONL file...")
+
+    try:
+
+        for commit in repo.iter_commits("HEAD"):
```

Comment:
```suggestion
        for commit in repo.iter_commits("--all"):
```
Looks like this gets passed down to `git rev-list` and we should export all the commits from the repo :ok_hand: 

For more info:

```sh
man git-rev-list
```